tracer/src/Datadog.Trace/IAST/IastModule.cs

```diff
@@ -77,12 +95,9 @@ public IastModule()
 
             tags.SetTag(Tags.Origin, IastOrigin);
```

Comment:
This looks like the wrong usage of the `_dd.origin` tag. That tag is a "trace-level" attribute meant to indicate which Datadog product _originated_ a trace, such as RUM, Synthetics, or CI Visibility. Even if you wanted to set the origin, you should _never_ overwrite an existing value, as is can break those products.

Reply:
It was added after this discussion: https://github.com/DataDog/dd-trace-dotnet/pull/3371#discussion_r999994887

In this case, we are setting the origin tag for a new span, so there is no danger to override the existing values. This is a new iast span and the only case where we set it. Still, if you think that the tag is wrong, we could discuss it and remove it.

Reply:
The point Lucas is making is that the origin _doesn't_ apply to a single span, it applies to _every_ span in a trace. So even if it's a new span, the origin may already be set for the trace, and this would overwrite it. That said ASM uses the origin tag as well, so I think this all needs clarification

Reply:
Ok, I see. Yes, I think that this would need clarification and the solution would affect both Iast and asm spans. I will discuss it with @robertpi as well.

Reply:
If it is ok, we will discuss it and I will add the needed changes in a new PR.

Reply:
Is this tag already required by the IAST backend? If not, I would prefer we _don't_ add it in this PR, and discuss what to do in a new PR.

As it is now, this tag is only expected in the first span of every trace chunk ( see[internal specs](https://datadoghq.atlassian.net/wiki/spaces/APM/pages/2357395856/Span+attributes#_dd.origin)). The trace agent also looks for the `_dd.origin` tag in the root span and copies it to the trace chunk metadata before sending it to the backend (see [source code](https://github.com/DataDog/datadog-agent/blob/56805dfa6935b3e5f682b758d5b4e240594b5124/pkg/trace/agent/normalizer.go#L163-L166)). Lastly, I'm about to merge #3316 which is making `Origin` a trace-level attribute on `TraceContext`.

You can still technically add this tag to any individual span, but the effect is unpredictable. For example, if that span happens to be the first span in a chunk, or what the trace agent considers a "chunk root", the origin could be applied to the entire chunk.

Reply:
It is not required. I can remove it and it will not have any bad consequence. I think that is fine to remove it now and discuss later if we want to add it again.

Reply:
I have removed the iast origin!

Reply:
Thanks!